src/lib.rs

```diff
@@ -601,6 +622,12 @@ mod opassign {
         }
     }
 
+    impl<T: Clone + NumAssign + MulAdd<Output = T>> MulAddAssign for Complex<T> {
+        fn mul_add_assign(&mut self, other: Complex<T>, add: Complex<T>) {
+            *self = self.clone().mul_add(other, add);
```

Comment:
Can this do any better using `T: MulAddAssign`?

Reply:
I don't think it possible for `impl for Complex<T>`

Reply:
I tried but I just get this error.
```rust
error[E0599]: no method named `mul_add` found for type `Complex<T>` in the current scope
   --> src/lib.rs:708:34
    |
83  | pub struct Complex<T> {
    | --------------------- method `mul_add` not found for this
...
708 |             *self = self.clone().mul_add(other.clone(), add.clone());
    |                                  ^^^^^^^
    |
    = note: the method `mul_add` exists but the following trait bounds were not satisfied:
            `&Complex<T> : traits::MulAdd<_, _>`
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `mul_add`, perhaps you need to implement one of them:
            candidate #1: `traits::Float`
            candidate #2: `traits::MulAdd`
            candidate #3: `traits::real::Real`
```

The only way I could solve it was with `MulAdd<Output = T>`.

Reply:
You'd also have to adjust the implementation, something like `self.re.mul_add_assign(...)` etc. - taking care to separate other operations that will need that original value.

It may not be any better, just something to try. We could also just leave ourselves open to this possibility in the function signature, by requiring both `MulAddAssign` and `MulAdd` even if we only use the latter for now.